node/cn/handler.go

```diff
@@ -465,14 +465,24 @@ func (pm *ProtocolManager) handle(p Peer) error {
 }
 
 func (pm *ProtocolManager) processMsg(msgCh <-chan p2p.Msg, p Peer, addr common.Address, errCh chan<- error) {
-	for msg := range msgCh {
-		if err := pm.handleMsg(p, addr, msg); err != nil {
-			p.GetP2PPeer().Log().Error("ProtocolManager failed to handle message", "msg", msg, "err", err)
-			errCh <- err
-			return
+	_, fakeF := pm.fetcher.(*fetcher.FakeFetcher)
+	_, fakeD := pm.downloader.(*downloader.FakeDownloader)
```

Comment:
Instead of changing this, How about modifying FakeFetcer/FakeDownloader.
The root cause is the FakerFetcher/Downloader was not designed well.
What do you think?